packages/ethash/src/index.ts

```diff
@@ -224,18 +223,31 @@ export default class Ethash {
         return cb(valid)
       }
 
-      async.eachSeries(
-        block.uncleHeaders,
-        (uheader: BlockHeader, cb2: (valid: boolean) => void) => {
-          this._verifyPOW(uheader, (valid3) => {
-            valid = valid3
-            cb2(valid)
+      const self = this
+
+      async function _verifyPowAsync() {
+        let error = false
+        for (let index = 0; index < block.uncleHeaders.length; index++) {
+          if (error) {
+            cb(false)
+          }
+          await new Promise((resolve, reject) => {
```

Comment:
can this code be cleaned up/flattened further? we can consider removing the callback and just supporting promises since ethash is mostly an internal lib just used in the blockchain package.

Reply:
Just took a look here: if we change this, this means we also have to "asyncify" the blockchain package. I agree it should be cleaned up and removed (we should be consistent and thus not have callback functions but instead promise functions in all packages of this monorepo; ethash and blockchain are exceptions currently). However I think this might be too much for this PR as this is likely to involves code changes for every call to the blockchain or ethash package (we thus have to convert every callback function call into a Promise function call).

If you are OK with that then I'll follow up with a new PR, if not then I'll build it on top of this one and we merge if that is ready.

Reply:
thanks, ok that sounds good to me. i'm fine with either integrating into this PR or opening a new one 👍

Reply:
Great! Could you approve this? Your other comment on ethash has the same remark and the nit one has been resolved 😄